sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventDirectorySteps.java

```diff
@@ -178,6 +199,50 @@ public EventDirectorySteps(
               FILTER_BY_TYPE_OF_PLACE, TypeOfPlace.getValueFor(typeOfPlace));
         });
 
+    When(
+        "I select Report Date among Event Reference Date options",
+        () -> {
+          webDriverHelpers.waitForPageLoaded();
+          webDriverHelpers.selectFromCombobox(
+              DATE_TYPE_COMBOBOX, EventReferenceDateOptions.REPORT_DATE.toString());
+        });
+
+    When(
+        "I fill in a date range",
+        () -> {
+          webDriverHelpers.waitForPageLoaded();
+          eventService.timeRange = buildTimeRange();
+          webDriverHelpers.fillInWebElement(
+              DATE_FROM_COMBOBOX,
+              eventService
+                  .timeRange[0]
+                  .format(DateTimeFormatter.ofPattern("dd.MM.yyyy"))
+                  .toString());
+          webDriverHelpers.fillInWebElement(
+              DATE_TO_COMBOBOX,
+              eventService
+                  .timeRange[1]
+                  .format(DateTimeFormatter.ofPattern("dd.MM.yyyy"))
+                  .toString());
+        });
+
+    When(
+        "I check that the dates of displayed Event results are correct",
+        () -> {
+          webDriverHelpers.waitForPageLoaded();
+          List<Map<String, String>> tableRowsData = getTableRowsData();
+          for (int i = 0; i < tableRowsData.size(); i++) {
+            String dateCell =
+                tableRowsData.get(i).get(EventsTableColumnsHeaders.REPORT_DATE_HEADER.toString());
+            LocalDate date = getLocalDateFromColumns(dateCell.substring(0, dateCell.indexOf(" ")));
+            softly.assertTrue(
+                date.isAfter(eventService.timeRange[0].minusDays(1))
+                    && date.isBefore(eventService.timeRange[1].plusDays(1)),
+                "The date(s) of displayed events are out of the requested range");
+            softly.assertAll();
```

Comment:
move the assertAll outside for loop, so you can collect all failures and check once the for loop is done